orchestrator/cli/domain_gen_helpers/product_block_helpers.py

```diff
@@ -23,35 +25,31 @@ def get_product_block_id(block_name: str) -> ScalarSelect:
 
 
 def get_product_block_ids(block_names: Union[List[str], Set[str]]) -> ScalarSelect:
-    return (
-        ProductBlockTable.query.where(ProductBlockTable.name.in_(block_names))
-        .with_entities(ProductBlockTable.product_block_id)
-        .scalar_subquery()
-    )
+    return select(ProductBlockTable.product_block_id).where(ProductBlockTable.name.in_(block_names)).scalar_subquery()
 
 
 def get_subscription_instance(subscription_id: str, product_block_id: ScalarSelect) -> ScalarSelect:
     return (
-        SubscriptionInstanceTable.query.where(
+        select(SubscriptionInstanceTable.subscription_instance_id)
+        .where(
             SubscriptionInstanceTable.product_block_id.in_(product_block_id),
             SubscriptionInstanceTable.subscription_id == subscription_id,
         )
-        .with_entities(SubscriptionInstanceTable.subscription_instance_id)
         .limit(1)
         .as_scalar()
     )
 
 
-def map_create_product_blocks(product_blocks: Dict[str, Type[ProductBlockModel]]) -> Dict[str, Type[ProductBlockModel]]:
+def map_create_product_blocks(product_blocks: dict[str, Type[ProductBlockModel]]) -> dict[str, Type[ProductBlockModel]]:
     """Map product blocks to create.
 
     Args:
         product_blocks: Dict of product blocks mapped by product block name.
 
     Returns: Dict of product blocks by product block name to create.
     """
-    _existing_product_blocks = ProductBlockTable.query.with_entities(ProductBlockTable.name).all()
-    existing_product_blocks = {block_name[0] for block_name in _existing_product_blocks}
+    _existing_product_blocks = db.session.scalars(select(ProductBlockTable.name))
+    existing_product_blocks = set(_existing_product_blocks)
```

Comment:
Does this work?

```suggestion
    existing_product_blocks = db.session.scalars(select(ProductBlockTable.name)).unique()
```

Reply:
Or perhaps [`from sqlalchemy import distinct`](https://docs.sqlalchemy.org/en/20/core/sqlelement.html#sqlalchemy.sql.expression.distinct) which should be even better; 

```suggestion
    existing_product_blocks = db.session.scalars(select(distinct(ProductBlockTable.name)))
```

Reply:
The `name` column on `product_blocks` is already unique by database index. So it's actually not even necessary to make a set out of it...